### DISCUSSION TITLE
Can the minio storage be set to private?

### DISCUSSION THREAD

**Discussion by TheZoker** [sorry-cypress/sorry-cypress#784] [Q&A]
Hi there,

I would like to set the minio storage access to private, so not everybody can upload any files to my public minio endpoint. 

Is there any way to archive this within sorry cypress (I mean it used the auth tokens already, right)?

Thanks!

**Comment by agoldis** [comment]
I don't think it's possible today because the read URL is publicly accessible. You would need to modify the code and send a signed read URL that wouldn't be accessible by 3rd parties. 
As an alternative, you can prevent access on a network level by using some kind of reverse proxy or firewall

**Reply by TheZoker** [reply]
Public read is fine, only write should be private (or protected by a token or something). 
As far as I can see, setting the bucket to public also allows anyone to upload files to the bucket without authentication, right?
![image](https://github.com/sorry-cypress/sorry-cypress/assets/1368405/d2334086-a21d-496f-b81e-ea226350fca3)


Ah ok so I would only allow connections from certain IPs to the bucket via my Nginx reverse proxy, right? 
Hmm but that would be difficult, since then the dashboard can no longer access the bucket, right? 

Edit: In the minio dashboard, can I create a new user (e.g. "sorry-cypress"), only give this user write permission to the bucket (read stays public), generate new access and secret key, set these in the docker compose file for the signed URL generation? Would that work? 

The policy could look like this:
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": "*",
            "Action": [
                "s3:GetObject"
            ],
            "Resource": [
                "arn:aws:s3:::sorry-cypress/*"
            ]
        },
        {
            "Effect": "Allow",
            "Principal": "*",
            "Action": [
                "s3:PutObject"
            ],
            "Resource": [
                "arn:aws:s3:::sorry-cypress/*"
            ],
            "Condition" : { "StringEquals" : { "aws:username" : "sorry-cypress" }}
        }
    ]
}

```
